File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
     return dfs(numbers, 0, 0, target)
}

function dfs (numbers, index, sum, target){
    if(index === numbers.length){
        if(sum === target) return 1;
        else return 0
    }
    
    return dfs(numbers, index + 1, sum + numbers[index], target) + dfs(numbers, index + 1, sum - numbers[index], target)
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(numbers, index, count, target) :
    if index == len(numbers) :
        if(target == count) :
            return 1
        else :
            return 0
    return dfs(numbers, index + 1, count + numbers[index], target) + dfs(numbers, index + 1, count - numbers[index], target)

def solution(numbers, target):
    answer = 0
    return dfs(numbers, 0, 0, target)